src/com/google/javascript/jscomp/ConvertIIFEArgsToVars.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2016 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.javascript.jscomp;
+
+import com.google.common.base.Preconditions;
+import com.google.javascript.rhino.IR;
+import com.google.javascript.rhino.Node;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Convert the arguments of an immediately invoked function expression to variables
```

Comment:
Add to the description:

This pass relies on variable renaming (done during normalization) to avoid accidental variable capture.